machine/minit.c

```diff
@@ -24,6 +24,19 @@ void* kernel_end;
 static void mstatus_init()
 {
   uintptr_t mstatus = 0;
+#if __riscv_xlen == 32
+  uint32_t mstatush = 0;
+#endif
```

Comment:
This presumably gives unused variable warnings for a little-endian RV32 build

Reply:
And if asserts are disabled? Why not just ifdef it out here for little-endian?

Reply:
It just seems not so clean to spread out the assumption that the RV32 register MSTATUSH can only be safely written on big endian systems (which is kind of a workaround in the first place) to multiple locations.  But that's just my personal view.